oak/server/rust/oak_runtime/src/channel.rs

```diff
@@ -239,6 +247,17 @@ impl DotIdentifier for ChannelId {
 impl Drop for Channel {
     fn drop(&mut self) {
         debug!("dropping Channel object {:?}", self);
+        match self.runtime_weak.upgrade() {
+            Some(runtime) => {
+                runtime.introspection_event(EventDetails::ChannelDestroyed(ChannelDestroyed {
+                    channel_id: self.id,
+                }));
+            }
+            None => {
+                warn!("Couldn't send ChannelDestroyed event, since the runtime has been dropped. {:?}", self);
+            }
```

Comment:
Given that this should never occur, I could also `unwrap` the result of the `upgrade`, but it seems safer to match & log a warning should the runtime ever happen to be dropped at this point.